src/main/webapp/dev/js/main/instructorCourseEnrollPage.js

```diff
@@ -73,89 +39,40 @@ const handsontable = new Handsontable(container, {
- */
-function updateHeaderOrder() {
-    const colHeaders = handsontable.getColHeader().join('|');
-    return colHeaders.concat('\n');
-}
-
-/**
- * Updates the student data from the spreadsheet when any of the
- * following event occurs afterChange, afterColumnMove or afterRemoveRow.
+ * Updates the student data from the spreadsheet when the user clicks "Enroll Students" button.
  *
  * Pushes the output data into the textarea (used for form submission).
- *
  */
 function updateDataDump() {
     const spreadsheetData = handsontable.getData();
-    let dataPushToTextarea = updateHeaderOrder();
-
-    for (let row = 0; row < spreadsheetData.length; row += 1) {
-        let countEmptyColumns = 0;
-        let rowData = '';
-        const spreadsheetDataRow = spreadsheetData[row];
-
-        for (let col = 0; col < spreadsheetDataRow.length; col += 1) {
-            rowData += spreadsheetDataRow[col] || '';
-            if (col < spreadsheetDataRow.length - 1) {
-                rowData += '|';
-                if (!spreadsheetDataRow[col]) {
-                    countEmptyColumns += 1;
-                }
-            }
-        }
-
-        if (countEmptyColumns < spreadsheetData[row].length - 1) {
-            dataPushToTextarea += rowData.concat('\n');
-        }
-    }
-
-    $('#enrollstudents').text(dataPushToTextarea);
-}
-
-/**
- * Adds the listener to specified hook name and only for this Handsontable instance.
- */
-function addHandsontableHooks() {
-    const hooks = ['afterChange', 'afterColumnMove', 'afterRemoveRow'];
-
-    hooks.forEach((event) => {
-        handsontable.addHook(event, updateDataDump);
-    });
+    const dataPushToTextarea = getUpdatedHeaderString(handsontable.getColHeader());
+    const userDataRows = getUserDataRows(spreadsheetData);
+    $('#enrollstudents').text(userDataRows === '' ?
+            '' : dataPushToTextarea + userDataRows); // prevent pushing header string if userDataRows is empty
 }
 
 $(document).ready(() => {
     prepareInstructorPages();
-    addHandsontableHooks();
 
     if ($('#enrollstudents').val()) {
-        const data = [];
-        const allData = $('#enrollstudents').val().split('\n'); // data in the table including column headers
-        const columns = allData[0].split('|');
-        const spreadsheetDataRows = allData.slice(1, -1);
+        const allData = $('#enrollstudents').val().split('\n'); // data in the table including column headers (string format)
+
+        const columnHeaders = allData[0].split('|');
+        handsontable.updateSettings({
+            colHeaders: columnHeaders,
+        });
 
+        const spreadsheetDataRows = allData.slice(1);
         if (spreadsheetDataRows.length > 0) {
-            for (let userDataRows = 0; userDataRows < spreadsheetDataRows.length; userDataRows += 1) {
-                data.push(spreadsheetDataRows[userDataRows].split('|'));
-            }
+            const data = getUpdatedData(spreadsheetDataRows);
```

Comment:
Actually when will the text area contain data in the enrollment page huh? Shouldn't it always be empty, cause it's for adding new students to the course.